lib/ios/TopBarAppearancePresenter.m

```diff
@@ -62,7 +66,15 @@ - (void)setLargeTitleAttributes:(RNNLargeTitleOptions *)largeTitleOptions {
     NSNumber* fontSize = [largeTitleOptions.fontSize getWithDefaultValue:nil];
     UIColor* fontColor = [largeTitleOptions.color getWithDefaultValue:nil];
     
-    _appearance.largeTitleTextAttributes = [RNNFontAttributesCreator createFromDictionary:_appearance.largeTitleTextAttributes fontFamily:fontFamily fontSize:fontSize defaultFontSize:nil fontWeight:fontWeight color:fontColor defaultColor:nil];
+    self.currentChildAppearance.largeTitleTextAttributes = [RNNFontAttributesCreator createFromDictionary:self.currentChildAppearance.largeTitleTextAttributes fontFamily:fontFamily fontSize:fontSize defaultFontSize:nil fontWeight:fontWeight color:fontColor defaultColor:nil];
+}
+
+- (UINavigationBarAppearance *)currentChildAppearance {
+    if (!self.boundViewController.childViewControllers.lastObject.navigationItem.standardAppearance) {
+        self.boundViewController.childViewControllers.lastObject.navigationItem.standardAppearance = [UINavigationBarAppearance new];
```

Comment:
1. I think ViewController should have a `getCurrentChild` function which will return `childViewControllers.lastObject` for parent VC's.
2. Lets add a `currentNavigationItem` to the presenter class to prevent the train wreck
3. `currentChildAppearance` is used as a getter - it should have the `get` prefix for consistency and readability.
4. Since `currentChildAppearance` is a getter, it shouldn't perform any logic other then getting a property. Especially, it shouldn't initialise values. this `getOrCreate` patters is very misleading and makes the code hard to reason about. If a `standardAppearance` is required for the correct functionality of the Presenter, then lets initialize it in the constructor.